spec/lib/prison_schedule_spec.rb

```diff
@@ -0,0 +1,158 @@
+      let(:friday) { Utilities::DAYS.fetch :friday }
+      let(:the_following_wednesday) { thursday + 6.days }
+
+      before { stub_const("PrisonDay::BANK_HOLIDAYS", [friday]) }
+
+      subject { described_class.new prison_not_working_weekends_with_three_lead_days }
+
+      it 'skips the public holiday' do
+        Timecop.travel(thursday) do
+          expect(subject.confirmation_email_date).to eq the_following_wednesday
+        end
+      end
+    end
+  end
+
+  describe '#available_visitation_dates' do
+    #      June 2015
+    # M  T  W  T  F  S  S
+    # 1  2  3  4  5  6  7
+    # 8  9  10 11 12 13 14
+    # 15 16 17 18 19 20 21
+    # 22 23 24 25 26 27 28
+    # 29 30 1  2  3  4  5
+
+    let(:one_off_anomalous_wednesday) { Date.parse 'Wed 10th June 2015' }
+    let(:unbookable_monday) { Date.parse 'Mon 15th June 2015' }
+    let(:unbookable_saturday) { Date.parse 'Sat 20th June 2015' }
+    let(:public_holiday) { Date.parse 'Mon 22nd June 2015' }
+
+    let(:prison_with_everyday_slots_except_wednesdays) do
+      prison_from prison_data.
+        merge(slots: slots_for_everyday.except('wed')).
+        merge(slot_anomalies: { one_off_anomalous_wednesday => "1000-1200" }).
+        merge(unbookable: [unbookable_monday, unbookable_saturday]).
+        merge(lead_days: 3)
+    end
+
+    let(:monday_june_01_2015) { Date.parse 'Mon 1st June 2015' }
+
+    let(:expected_date_range) { Date.parse('Fri 5th June 2015')..Date.parse('Fri 3rd July 2015') }
+
+    let(:expected_dates) do
+      expected_date_range.to_a.delete_if(&:wednesday?).append(one_off_anomalous_wednesday) -
+        [public_holiday, unbookable_monday, unbookable_saturday]
+    end
+
+    before { stub_const("PrisonDay::BANK_HOLIDAYS", [public_holiday]) }
+
+    around { |example| Timecop.travel(monday_june_01_2015) { example.run } }
+
+    subject { described_class.new prison_with_everyday_slots_except_wednesdays }
+
+    it 'returns a list of days the prison can accept visitors excluding unbookable dates & public holidays' do
+      expect(subject.available_visitation_dates).to match_array expected_dates
+    end
+
+    it 'ignores days that are not specified as slots unless a date is anomalous' do
+      wednesday_slots = subject.available_visitation_dates.select(&:wednesday?)
+      expect(wednesday_slots).to match_array [one_off_anomalous_wednesday]
+    end
+
+    describe 'the range of dates covered' do
+      let(:the_day_after_the_conformation_email) { subject.confirmation_email_date.tomorrow }
+      let(:booking_window_days) { 28.days }
+      let(:the_last_bookable_day) { the_day_after_the_conformation_email + booking_window_days }
+
+      it 'starts from the day after the confirmation email' do
+        expect(subject.available_visitation_dates.first).to eq the_day_after_the_conformation_email
+      end
+
+      it 'the end date is equivalent to the number of booking window days for the prison plus the confirmation email date' do
```

Comment:
How about calling this `it "doesn't offer booking windows past the number of bookable days after the confirmation email date"`?